File: MyProjects/MAP/Assignment 7/Interpreter/Model/Expressions/VarExpr.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Interpreter.Model.Expressions
{
    public class VarExpr:IExpr
    {
        private string id;

        public VarExpr(string v)
        {
            id = v;
        }

        public int evaluate(Utilities.IDictionary<string, int> dic)
        {
            return dic.getValue(id);
        }

        public override string ToString()
        {
            return id;
        }
    }
}

File: MyProjects/MAP/Practical Exam/ConditionalStmt/src/Model/Expressions/VarExpr.java
package Model.Expressions;

import Model.Utilities.IDictionary;

public class VarExpr implements IExpr {
    private String id;

    public VarExpr(String v){
        id=v;
    }
    @Override
    public int evaluate(IDictionary<String,Integer> dic, IDictionary<Integer, Integer> heap){
        return dic.getValue(id);
    }

    public String toString(){
        return id;
    }
}
